mrp_stock_analytic/views/mrp_production_views.xml

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="mrp_production_form_view" model="ir.ui.view">
+        <field name="name">mrp.production.form</field>
+        <field name="model">mrp.production</field>
+        <field name="inherit_id" ref="mrp_account.mrp_production_form_view_inherited" />
+        <field name="arch" type="xml">
+            <field name="analytic_distribution" position="attributes">
+                <attribute name="readonly">state in ('done', 'cancel')</attribute>
+            </field>
+        </field>
+    </record>
+    <record id="mrp_production_tree_view" model="ir.ui.view">
+        <field name="name">mrp.production.tree</field>
+        <field name="model">mrp.production</field>
+        <field name="inherit_id" ref="mrp.mrp_production_tree_view" />
+        <field name="arch" type="xml">
+            <field name="analytic_distribution" position="attributes">
+                <attribute name="readonly">state in ('done', 'cancel')</attribute>
+                <attribute name="optional">show</attribute>
```

Comment:
Is there a compelling reason to override upstream's optional="hide"? It is also linked to the analytic accounting user group there. https://github.com/odoo/odoo/commit/f13d537a42ff1a2713484d0e7c39b73fbb87a7fb#diff-cd75111aeb0015ab66e8b516edbfc92044f852718ec73251f3fdbfee1b69cfbeR11

Reply:
Can't see one. It was originally added with the 16 version of the module, so maybe the field wasn't added yet at that point? I removed our override in https://github.com/OCA/account-analytic/pull/644/commits/d2f3771f2e00b4b77ba792034e26cb3465fe980f.

Reply:
Yes indeed, there was no analytic distribution in Odoo 16's mrp.production model. As such, the last two commits (maybe other commits as well?) can be squashed into the migration commit (and I will try very hard to make this my last comment :grin:)

Reply:
Alright, squashed those commits into the migration commit, and the final test-fix into the earlier commit that added the test. I feel like the current set of commits are now separate changes, and I personally wouldn't squash them any further.

Reply:
Thank you, and agreed!